pip/configuration.py

```diff
@@ -24,12 +24,22 @@
 )
 from pip.utils import ensure_dir, enum
 
+if False:
+    from typing import Any, Dict, Iterable, List, NewType, Optional, Tuple
+
+    try:
+        from configparser import ConfigParser
+    except ImportError:
+        from ConfigParser import ConfigParser  # type: ignore
+
```

Comment:
This is so that we don't need to actually import typing at runtime (which is usually not a problem but pip, unsurprisingly, is a special case) 

We can't use typing without vendoring it, which is not possible due to the fact that it uses different sources for Py2/Py3). The next best thing is using utilizing the fact that typing import is kinda optional - https://github.com/python/mypy/issues/3216 - which is what this is.

---

May I add a comment saying "just mypy things" and a link to the GH issue? :3